components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3194,11 +2903,662 @@ subroutine rain_immersion_freezing(t,    &
+
+real(rtype), intent(out) :: qcacc
+real(rtype), intent(out) :: ncacc  
+
+real(rtype) :: dum, dum1 
+
+if (qr_incld.ge.qsmall .and. qc_incld.ge.qsmall) then
+
+   if (iparam.eq.1) then
+      !Seifert and Beheng (2001)
+      dum   = 1._rtype-qc_incld/(qc_incld+qr_incld)
+      dum1  = (dum/(dum+5.e-4_rtype))**4
+      qcacc = kr*rho*0.001_rtype*qc_incld*qr_incld*dum1
+      ncacc = qcacc*rho*0.001_rtype*(nc_incld*rho*1.e-6_rtype)/(qc_incld*rho*   &
+           0.001_rtype)*1.e+6_rtype*inv_rho
+   elseif (iparam.eq.2) then
+      !Beheng (994)
+      qcacc = 6._rtype*rho*(qc_incld*qr_incld)
+      ncacc = qcacc*rho*1.e-3_rtype*(nc_incld*rho*1.e-6_rtype)/(qc_incld*rho*1.e-3_rtype)* &
+           1.e+6_rtype*inv_rho
+   elseif (iparam.eq.3) then
+      !Khroutdinov and Kogan (2000)
+      qcacc = 67._rtype*(qc_incld*qr_incld)**1.15_rtype
+      ncacc = qcacc*nc_incld/qc_incld
+   endif
+
+   if (qcacc.eq.0._rtype) ncacc = 0._rtype
+   if (ncacc.eq.0._rtype) qcacc = 0._rtype
+
+endif
+
+end subroutine cloud_rain_accretion 
+
+subroutine rain_self_collection(rho,qr_incld,nr_incld,    &
+   nrslf)
+   
+   !.....................................
+   ! self-collection and breakup of rain
+   ! (breakup following modified Verlinde and Cotton scheme)
+
+   implicit none 
+   
+   real(rtype), intent(in) :: rho 
+   real(rtype), intent(in) :: qr_incld
+   real(rtype), intent(in) :: nr_incld
+   real(rtype), intent(out) :: nrslf 
+
+   real(rtype) :: dum, dum1, dum2 
+
+   if (qr_incld.ge.qsmall) then
+
+      ! include breakup
+      dum1 = 280.e-6_rtype
+
+      ! use mass-mean diameter (do this by using
+      ! the old version of lambda w/o mu dependence)
+      ! note there should be a factor of 6^(1/3), but we
+      ! want to keep breakup threshold consistent so 'dum'
+      ! is expressed in terms of lambda rather than mass-mean D
+
+      dum2 = (qr_incld/(pi*rhow*nr_incld))**thrd
+      if (dum2.lt.dum1) then
+         dum = 1._rtype
+      else if (dum2.ge.dum1) then
+         dum = 2._rtype-exp(2300._rtype*(dum2-dum1))
+      endif
+
+      if (iparam.eq.1) then
+         nrslf = dum*kr*1.e-3_rtype*qr_incld*nr_incld*rho
+      elseif (iparam.eq.2 .or. iparam.eq.3) then
+         nrslf = dum*5.78_rtype*nr_incld*qr_incld*rho
+      endif
+
+   endif
+
+end subroutine rain_self_collection
+
+
+subroutine cloud_water_autoconversion(rho,inv_rho,qc_incld,nc_incld,qr_incld,mu_c,nu,    &
+   qcaut,ncautc,ncautr)
+
+   implicit none 
+
+   real(rtype), intent(in) :: rho
+   real(rtype), intent(in) :: inv_rho
+   real(rtype), intent(in) :: qc_incld
+   real(rtype), intent(in) :: nc_incld 
+   real(rtype), intent(in) :: qr_incld
+   real(rtype), intent(in) :: mu_c 
+   real(rtype), intent(in) :: nu 
+
+   real(rtype), intent(out) :: qcaut
+   real(rtype), intent(out) :: ncautc 
+   real(rtype), intent(out) :: ncautr
+
+   logical :: qc_not_small 
+   real(rtype) :: dum, dum1 
+
+   qc_not_small: if (qc_incld.ge.1.e-8_rtype) then
```

Comment:
It probably won't be BFB, but what would happen if we switched the criteria for `qc_not_small` from `1e-8` to `qsmall`?  Is there a reason why the cutoff is `1e-8`?

Reply:
Let's delay this non-B4B change so we can keep the current PR a refactor. I created an issue so that we deal with this later. (#121 (comment))